src/feature/rend/rendservice.c

```diff
@@ -130,6 +130,22 @@ static smartlist_t *rend_service_list = NULL;
  * service on config reload. */
 static smartlist_t *rend_service_staging_list = NULL;
 
+/** Helper: log the deprecation warning for version 2 only once. */
+static void
+log_once_deprecation_warning(void)
+{
+  static bool logged_once = false;
+  if (!logged_once) {
+    log_warn(LD_REND, "DEPRECATED: Onion service version 2 are deprecated. "
+             "Please use version 3 which is the default now. "
+             "Currently, version 2 is planned to be obsolete in "
+             "the Tor version 0.4.6 stable series.");
```

Comment:
Do you think we should link to the tor-dev post or the blog post here?

Reply:
As discussed, maybe not ideal to have links hardcoded in our tor codebase as in the link then has to be a super long permalink...